src/test/java/com/google/devtools/build/lib/skyframe/RecursiveFilesystemTraversalFunctionTest.java

```diff
@@ -674,6 +684,22 @@ private void assertTraversalOfDirectory(Artifact directoryArtifact) throws Excep
     assertThat(v4).isEqualTo(v3);
     assertTraversalRootHashesAreEqual(v3, v4);
     assertThat(progressReceiver.invalidations).doesNotContain(traversalRoot);
+
+    // Add a new empty subdirectory to the directory and see that the value is rebuilt, but results
+    // in the same hash.
+    if (!directoryArtifact.isTreeArtifact()) {
```

Comment:
Why not for tree artifact?

Reply:
I didn't know how to add an empty directory to a tree artifact that doesn't seem artifical -  after all, there is `TreeEmptyDirectoryArtifact` (yet). I could add a `TreeFileArtifact` for the empty directory instead, but would testing this even make sense?

Reply:
Great point, maybe add a comment about that.

Reply:
Added a comment.